snakemake/executors/slurm/slurm_submit.py

```diff
@@ -0,0 +1,349 @@
+        content = self.format_job(self.jobscript, job, exec_job=exec_job, **kwargs)
+
+        logger.debug("Jobscript:\n{}".format(content))
+        with open(jobscript, "w") as f:
+            print(content, file=f)
+
+        os.chmod(jobscript, os.stat(jobscript).st_mode | stat.S_IXUSR | stat.S_IRUSR)
+
+    def run(self, job, callback=None, submit_callback=None, error_callback=None):
+        super()._run(job)
+        workdir = os.getcwd()
+        jobid = job.jobid
+        # generic part of a submission string:
+        # print(self.exec_job)
+        os.makedirs('.snakemake/slurm_logs', exist_ok=True)
+
+        try:
+            call = "sbatch -A {account} -p {partition} \
+                    -J {jobname} \
+                    -o .snakemake/slurm_logs/%x_%j.log \
+                    --export=ALL".format(
+                **job.resources, jobname=self.jobname
+            )
+        except KeyError as e:
+            logger.error(
+                "Missing job submission key '{}' for job '{}'.".format(
+                    e.args[0], job.name
+                )
+            )
+            sys.exit(1)
+        if not job.resources.get('walltime_minutes'):
+            logger.warning("No wall time limit is set, setting 'walltime_minutes' to 1.")
+        call += " -t {walltime_minutes}".format(walltime_minutes = job.resources.get('walltime_minutes', default_value=1))
+        if job.resources.get("constraint"):
+            call += " -C {constraint}".format(**job.resources)
+        # TODO: implement when tempfs-resource is defined
+        # if job.resources.get()
+        if job.resources.get("mem_mb_per_cpu"):
+            call += " --mem-per-cpu={mem_mb_per_cpu}".format(**job.resources)
+        elif job.resources.get("mem_mb"):
+            call += " --mem {mem_mb}".format(**job.resources)
+        else:
+            logger.warning(
+                "No job memory information ('mem_mb' or 'mem_mb_per_cpu') is given - submitting without. This might or might not work on your cluster."
+            )
+
+        # if ressources == MPI:
+        if job.resources.get("mpi", False):
+            # submit MPI job
+            # mold submissing string, submit according to generic cluster exec
+            pass
+        # ordinary smp application:
+        elif not job.resources.get("mpi") and not job.is_group():
+            # TODO: this line will become longer
+            # TODO: hence the single command, yet
+            if not job.resources.get("threads"):
+                logger.warning("No value for 'threads' is set, assuming 1.")
+                call += " -n 1 -c 1"
+            else:
+                call += " -n 1 -c {threads}".format(**job.resources)
+        else:  # job.is_group:
+            pass
+
+        # as we cannot do 'sbatch ... cd {workflow.workdir_init} && python ...' as snakemake otherwise expects
+        call += " --chdir={workdir_init}".format(
+            workdir_init=self.workflow.workdir_init
+        )
```

Comment:
The `cd ...` happens inside of the jobscript. Hence, this should not be an issue I think, and this additional code could be omitted?